plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -243,15 +232,46 @@ fun reauthConnectionIfNeeded(
     project: Project?,
     connection: ToolkitConnection,
     onPendingToken: (InteractiveBearerTokenProvider) -> Unit = {},
+    isReAuth: Boolean = false
 ): BearerTokenProvider {
     val tokenProvider = (connection.getConnectionSettings() as TokenConnectionSettings).tokenProvider.delegate as BearerTokenProvider
     if (tokenProvider is InteractiveBearerTokenProvider) {
         onPendingToken(tokenProvider)
     }
 
     maybeReauthProviderIfNeeded(project, tokenProvider) {
-        runUnderProgressIfNeeded(project, AwsCoreBundle.message("credentials.pending.title"), true) {
-            tokenProvider.reauthenticate()
+        runUnderProgressIfNeeded(project, message("credentials.pending.title"), true) {
+            try {
+                tokenProvider.reauthenticate()
+                if (isReAuth) {
+                    recordLoginWithBrowser(
+                        credentialStartUrl = tokenProvider.currentToken()?.ssoUrl,
+                        credentialSourceId = CredentialSourceId.AwsId,
+                        isReAuth = true,
+                        result = Result.Succeeded
+                    )
+                    recordAddConnection(
+                        credentialSourceId = CredentialSourceId.AwsId,
+                        isReAuth = true,
+                        result = Result.Succeeded
+                    )
```

Comment:
i still dont understand why these are separate metrics

Reply:
addConnection is to be used in other places too where the call fails (eg: prior to the browser step)

Reply:
AddConnection covers entire flow. 

- Moment users clicks "add new Connection" we record this metric, meaning if they give up only looking at auth options (idc, bId or iam), we emit auth_addConnection with result:Cancelled. 

- if they have chosen one of the auth options and leave when they invalid starturl, profile name,etc we emit auth_addConnection {invalidInputFields}. 

- After all this when user enters auth flow we emit loginWithBrowser to capture nuances of browser flow. we also miss users loggin with IAM creds in loginWithBrowser so auth_addConnection helps us give full picture. 

There are some scenarios where user dont go through normal auth flow and skips to brower, such as Black Caimen users, reauth in this case we rely on loginWithBrowser to complete auth picture. 

more details on design: https://quip-amazon.com/M2VdAWUanrLv/New-Auth-telemetry-design